Update LTLI

aphillips opened the pull request:
- Make document consistent with Note status by removing contents about "normative"
- Added local.css (copied from string-meta)
- Coped string-meta's best practices note
- Styled, organized, and extended the best practices
- Added definitions for valid and well-formed
- Fixed my attribution
- Fixed respec errors and made references us https
- Moved the language tag terminology section under the I18N terminology section
- A few other edits to abstract, introduction

This is the first in what is expected to be a series of edits to bring this document up to Note status.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/ltli/pull/6.html" title="Last updated on Jul 7, 2020, 2:09 PM UTC (558e13b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/ltli/6/b9b5ddf...aphillips:558e13b.html" title="Last updated on Jul 7, 2020, 2:09 PM UTC (558e13b)">Diff</a>

aphillips committed:
Converting to Note status by removing all conformance and normative/non-normative section labels.
Fix various respec errors.
Correct author attributions.
Moved Section defining language tags below core terminology (i18n, etc.) section.
Removed extra section layers.

TODO: fix (by removing) the conformance section. Additional writing/revisions. etc.

aphillips committed:
Revised abstract and part of the introduction.
Added a dfn around the term "natural language"

aphillips committed:
Added local.css (a copy of the one in string-meta)

aphillips committed:
Copied string-meta's conformance note.
Added best practices related to obs-language-tag.
Added best practices related to valid and well-formed.
Added definitions of valid and well-formed.

aphillips committed:
Moved valid/well-formed to terminology section. Adjusted best practices section to match.

aphillips commented:
Note: Felix (@fsasaki) isn't listed on this repo still. @r12a or @xfq can you add him to reviewers?

aphillips committed:
Fixed one editorial inconsistency ("Because...")

aphillips committed:
Edits to fix appearance of Appendix A.
Edits to mention Unicode locales in introduction.

xfq committed:
Merge branch 'gh-pages' into gh-pages

xfq commented:
> Note: Felix (@fsasaki) isn't listed on this repo still. @r12a or @xfq can you add him to reviewers?

Done.

(BTW, I have resolved the merge conflict introduced by #5.)

xfq commented on index.html:
```diff
@@ -81,132 +72,28 @@
+        
+        <p>Historically, locales were identified by the programming language or operating environment of the user. This application-specific identifier was often inferred from language tags. For example, an implementation could map a language tag from an existing protocol, such as HTTP's Accept-Language header, to its locale model. Locales may also be identified directly by using the language tag ...
+        
+        <p>Unicode Locales are an extension of [[BCP47]] defined as part of Unicode's [[CLDR]] project. These identifiers are identical to language tags, but apply additional rules about the content of certain language tags. Unicode Locales increasingly form the basis for <a>internationalization</a> on the Web, particularly as part of the <code>Intl</code> locale framework in JavaScript.</p>
```
We should probably add a link to the definition of "Unicode Locales" here (i.e., `<a>Unicode Locales</a>`). And it is "Unicode locales" (lowercase) in the definition in section 2.

aphillips replied:
Fixed in 645ee75

xfq commented on index.html:
```diff
@@ -308,62 +195,145 @@ <h3>What are Internationalization and Localization?</h3>
+          is a collection of one or more language ranges identifying the user's
+          language preferences for use in matching. As the name suggests, such
+          lists are normally ordered or weighted according to the user's
+          preferences. The HTTP [[RFC2616]] <code>Accept-Language</code>[[RFC3282]]
```
```suggestion
          preferences. The HTTP [[RFC2616]] <code>Accept-Language</code> [[RFC3282]]
```

xfq commented on index.html:
```diff
@@ -308,62 +195,145 @@ <h3>What are Internationalization and Localization?</h3>
+        <p>This section provides specification authors and implementers with best practices recommended by the Internationalization (I18N) Working Group. These (and many other) best practices, along with links to supporting materials, can also be found in the <cite>Internationalization Best Practices for Spec Developers</cite> [[INTERNATIONAL-SPECS]]. In addition to the best practices found here, ...
+        
+        <aside class="note">
+           <p>In this section [[RFC2119]] keywords have their usual meaning. We differentiate <em>best practices</em>, which should be adopted by all specifications and <em>recommendations</em>, which require additional standardization or which are speculative prior to adoption.</p>
```
Should we update the reference to BCP 14? (See an example here: https://www.w3.org/TR/appmanifest/#conformance )

aphillips replied:
Respec does not know  BCP14. I can add a bibxml reference to our document later. I filed an issue to track it.

xfq commented on index.html:
```diff
@@ -308,62 +195,145 @@ <h3>What are Internationalization and Localization?</h3>
+        <p class="advisement">Specifications for the Web that require language identification MUST refer to [[BCP47]]. </p>
+        
+        <p class="advisement">Specifications SHOULD NOT refer to specific component RFCs.</p>
+        <p>The "BCP" nomenclature refers to the current set of RFCs that form the "best current practice". At the time this document was published, [[BCP47]] consisted of two RFCs: <cite>Tags for the Identification of Languages </cite>[[RFC5646]] and <cite>Matching of Language Tags</cite>[[RFC4647]].</p>
```
```suggestion
        <p>The "BCP" nomenclature refers to the current set of RFCs that form the "best current practice". At the time this document was published, [[BCP47]] consisted of two RFCs: <cite>Tags for the Identification of Languages </cite>[[RFC5646]] and <cite>Matching of Language Tags</cite> [[RFC4647]].</p>
```

aphillips committed:
Update index.html

Co-authored-by: Fuqiao Xue <xfq@w3.org>

aphillips committed:
Update index.html

Co-authored-by: Fuqiao Xue <xfq@w3.org>

aphillips committed:
Added definitions for Unicode locale. Rewriting many parts of the locale/i18n/l10n definition section to account for these changes.

aphillips committed:
Fix the BCP14 reference back to RFC2119 because it makes respec barf.

aphillips committed:
Merge branch 'gh-pages' of https://github.com/aphillips/ltli into gh-pages

aphillips committed:
Revisions to introduction to address @xfq's comment. Basically I removed the forward definition of Unicode locale (and all of the associated text--moving a relevant bit of it into the section below). I then added text that made clear the purpose of this document.

I also added a reference to string-meta.

fsasaki commented on index.html:
```diff
@@ -11,7 +11,7 @@
+                    company:    "Amazon.com"
 	    },
 	              {   name: "Felix Sasaki",
 		                company: "W3C"
```
Can you change my affiliation to "invited expert"?

aphillips replied:
Done.

fsasaki commented on index.html:
```diff
@@ -81,100 +72,125 @@
+        
+        <p><a>Language tags</a> can also be used to identify <a>international preferences</a> associated with a given piece of content or user because these preferences are linked to the natural language, regional association, or culture of the end user. Such preferences are applied to processes such as presenting numbers, dates, or times; sorting lists linguistically; providing defaults for item ...
+        
+        <p>This document describes best practices for the adoption and use of [[BCP47]] language tags for the identification of <a>natural language</a> content as well as the use of language tags to represent the locale preferences of the user or content author. It describes how document formats, specifications, and implementations should handle the language tags described by [[BCP47]], as well a ...
```
Since later in the LTLI draft, there is a definition of international preferences, maybe the reference to WS-I18N-SCENARIOS should be moved to that section, or even dropped?

aphillips replied:
Good catch. I removed the reference, since my intention is for LTLI to be the "standard bearer" for these definitions going forwards.

fsasaki commented on index.html:
```diff
@@ -207,163 +223,65 @@ <h3>Languages, Language Tags and Matching of Language Tags</h3>
-            'valid'.</p>
-        </div>
+          
+        <p class="advisement">Specifications MUST NOT reference obsolete versions of [[BCP47]], such as [[RFC1766]] or [[RFC3066]].</p>
```
General comment: the "MUST, "MUST NOT" and "SHOULD NOT" statements do not have IDs. Should these be added, to make referencing such statements later easier?

aphillips replied:
Definitely. I added IDs.

fsasaki commented on index.html:
```diff
@@ -516,25 +430,14 @@ <h5>Use of <code>mul</code></h5>
-      <p>Further application scenarios of this specification encompass for
-        example the standards mentioned in <a href="#scope">Scope of this
-          Specification</a>. The scenarios can be divided in four areas:</p>
+      <p class="issue">This section was part of the original LTLI document and exists to address items identified in [[WS-I18N-REQ]]. It refers to this document as a "specification" (which it no longer is). Removing this section is probably the next step.</p>
```
+1 to removing this section.

aphillips replied:
Done

r12a commented on index.html:
```diff
@@ -81,100 +72,125 @@
+        
+        <p>Identification of language and locale has a broad range of applications within the World Wide Web. Existing standards which make use of language identification include the <code>xml:lang</code> attribute in [[XML10]], the <code>lang</code> and <code>hreflang</code> atttributes in [[HTML]], the <code>language</code> property in [[XSL10]], and the <code>:lang</code> pseudo-class in CSS [ ...
+ 
+        <p>This document's focus is on the interplay between <a>language tags</a> and the <a>internationalization</a> of the Web. Many applications need to provide for the <a>localized</a> presentation of data values (such as numbers or dates). This document provides <a href="#best-practices">best practices</a> for specification authors who need to define language tags in their document formats o ...
```
Reading through this, i get the feeling that i'm reading this paragraph for the 4th time. Maybe ok to leave for now, but perhaps later boil this section down a bit, especially the first sentences of each paragraph.

aphillips replied:
That makes sense.

aphillips replied:
Upon further reflection, I removed the sentence in question.

aphillips committed:
Address Felix's comments.
- changed his attribution
- removed the ws-i18n reference
- removed the ws-i18n appendix
- added IDs for all BPs

In addition, I moved the remaining requirements to the bp section and commented out the blocks at the bottom.

aphillips committed:
Address Richard's comment by removing the offending sentence.
Added a reference to ECMA402.
Converted an issue into a "Further Reading" section.

aphillips committed:
Further condensation of the introduction per @r12a
Fix a typo.

aphillips committed:
One more edit to clarify and reduce repetition.

aphillips committed:
Fixed more references and data-lt attributes.
Further work on the introduction.

aphillips committed:
Added reference styling to local style sheet and visible reference IDs.
Added reference to ECMA262 ("JavaScript")

xfq commented on index.html:
```diff
@@ -1,38 +1,55 @@
+		    "Isaac Durazo"
+        ]},
+        
+    "ECMA262": {
```
You can use `[ECMASCRIPT]` to refer to ECMA-262 in ReSpec (without adding a custom entry in `localBiblio`).

aphillips replied:
Good catch. Fixed.

xfq commented on index.html:
```diff
@@ -1,38 +1,55 @@
-localBiblio:  {
+    localBiblio:  {
+
+	"ECMA402": {
```
You can use `[ECMA-402]` to refer to ECMA-402 in ReSpec (without adding a custom entry in `localBiblio`).

aphillips replied:
Good catch. Fixed.

aphillips committed:
Removed remaining extra sections.
Added links to the BP ids and styles to show them.

aphillips commented:
Given all the changes, I'm going to merge this PR.
